### ISSUE TITLE
VS 2015 Consolidate feature at solution level UX

### ISSUE THREAD

**Issue by brgrz** [NuGet/Home#1987]
When using solution level consolidate feature in Visual Studio 2015 Update 1 the behaviour is as follows:
- if we have multiple packages to consolidate, select one then select the checkboxes for the desired projects for which to consolidate on the right side of the window,
- hit Install and it will update the packages across projects,
- then switch to another package within the list and the previous checkboxes remain checked.

It would make more sense if checkboxes either cleared or the required ones would get checked automatically. As it is now, totally irrelevant projects remain checked for consolidation.

I don't know whether this belongs here or is more Visual Studio related.

**Comment by csharpfritz** [comment]
What makes a project "totally irrelevant"?  That's a very opinionated
statement.

I could see a discussion around clearing or marking all checkboxes when
selecting new packages.  However, in this case it allows you to bring a new
project in a solution up to pace with the correct version of packages that
are already in use in other projects.

Is that the best use of this functionality?  Perhaps not, but I am not
convinced that this functionality needs to change in the short term.

**Comment by yishaigalatzer** [comment]
I believe we fixed this one already in 3.4. Assigning to @RanjiniM to validate

**Comment by yishaigalatzer** [comment]
BTW the original design was to keep the projects list checked so when you switch between projects (assuming say you removed one project, we don't auto select it for you again, we realized a bit late that it is not the super common path)

**Comment by brgrz** [comment]
Yeah I agree it is more common consolidating the existing projects rather than bringing new ones up to date with packages. As I described it is all the way most software works when switching between unrelated options which packages on the left side in that dialog often are.

**Comment by zihotki** [comment]
According to my experience the feature is mostly relevant for old projects. But current UI requires to do a lot of clicks to consolidate a single package across many projects (imagine a solution with 80 projects and you need to consolidate package in 30 of them, good luck with selecting all of them).

**Comment by yishaigalatzer** [comment]
@zihotki the idea is that you can sort by version or project, so you can idealy click through what you want.

We should probably do more thinking here, and improve. This was a feature we had a short amount of time to invest in, and we could invest more going forward. 

This feature enables installing packages into additional project (was a specific ask by many customers), so what could be seemingly "weird" behavior is useful to many.

Things I took for this that we should consider
1. We could look at auto selection (when you switch between packages, we select all the projects that have them) - That should probably be an option/toggle
2. We could add an auto select button
3. We could clear the selection/and remember previous selection for each package.

I'm leaning towards option 1, but we will have to spend more time designing and playing with it before we can lock it down :)

**Comment by zihotki** [comment]
@yishaigalatzer IMO either 1 or 2 will work fine. Also the UI could support batch selection of projects - select multiple projects in UI and press Space to check them all. That will be a bit more work but still manageble. Unfortunately the latest version of Package Manager for VS2015 doesn't check all selected projects but only the last one you selected. Probably a new issue should be logged for this, I'm not sure.

**Comment by emgarten** [comment]
Thank you very much for your feedback. We appreciate you for letting us know about the problem you encountered, but we haven’t been able to fully investigate your problem as our focus has been on other problem reports. If there are more community feedback on this, then we'll re-evaluate this.
